tensorbay/opendataset/PlantSeedlings/loader.py

```diff
@@ -0,0 +1,175 @@
+                1.png
+                2.png
+                ...
+            Charlock/
+            Cleavers/
+            Common Chickweed/
+            Common wheat/
+            Fat Hen/
+            Loose Silky-bent/
+            Maize/
+            Scentless Mayweed/
+            Shepherd’s Purse/
+            Small-flowered Cranesbill/
+            Sugar beet/
+            Segmented/
+                Black-grass/
+                 1.png
+                 2.png
+                 ...
+                Charlock/
+                Cleavers/
+                Common Chickweed/
+                Common wheat/
+                Fat Hen/
+                Loose Silky-bent/
+                Maize/
+                Scentless Mayweed/
+                Shepherd’s Purse/
+                Small-flowered Cranesbill/
+                Sugar beet/
+
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    for subset in os.listdir(root_path):
+        if subset == "Segmented":
+            continue
+        segment = (
+            _get_segment(root_path, subset)
+            if subset != "Fullimages"
+            else _get_fullimages(root_path)
+        )
+        dataset.add_segment(segment)
+    return dataset
+
+
+def _get_fullimages(root_path: str) -> Segment:
+    segment = Segment("Fullimages")
+    files_path = os.path.join(root_path, "Fullimages")
+    category_map = _get_category_map()
+    for file_path in glob(os.path.join(files_path, "[0-9]*")):
+        file = os.path.basename(file_path)
+        for data in _generate_fullimages_data(os.path.join(files_path, file), category_map[file]):
+            segment.append(data)
+    return segment
+
+
+def _get_category_map() -> Dict[str, str]:
+    category_map = {}
+    for key, value in _FULLIMAGES_MAP.items():
+        for index in range(key[0], key[1] + 1):
+            category_map[str(index)] = value
+    return category_map
+
+
+def _generate_fullimages_data(data_path: str, category: str) -> Iterator[Data]:
+    for data_subset in os.listdir(data_path):
+        data = Data(
+            os.path.join(data_path, data_subset), target_remote_path=f"{category}_{data_subset}"
+        )
+        data.label.classification = Classification(category=category)
+        yield data
+
+
+def _get_segment(root_path: str, subset: str) -> Segment:
+    segment = Segment(subset.replace(" ", "_"))
+    data_path = os.path.join(root_path, subset)
+    segmented_path = os.path.join(root_path, "Segmented", subset)
+    single_channel_segmented_path = os.path.join(segmented_path, "single_channel_segmented")
+    if not os.path.exists(single_channel_segmented_path):
+        os.makedirs(single_channel_segmented_path)
```

Comment:
Is it better to make a "single_channel_segmented" folder at the same level as the "Segmented" folder?

Reply:
I don't think so, it will break the original file path structure, and I need to pass one more parameter